lectures/sympy.md

```diff
@@ -0,0 +1,994 @@
+
+$$
+PV_{\text{{highschool}}} = \sum_{t=0}^T R^{-t} w_t^h
+$$
+
+It is the sum of the discounted earnings from the first year after high school to the last year of work.
+
+```{code-cell} ipython3
+PV_college = Sum(R**-t * w_ct, (t, 4, T))
+PV_college
+```
+
+```{code-cell} ipython3
+PV_highschool = Sum(R**-t * w_ht, (t, 0, T))
+PV_highschool
+```
+
+We can evaluate the sum using the `doit` method
+
+```{code-cell} ipython3
+PV_hs = simplify(PV_highschool.doit())
+PV_hs
+```
+
+```{code-cell} ipython3
+PV_c = simplify(PV_college.doit())
+PV_c
+```
+
+### Defining the Indifference Equation
+
+Now we introduce the symbol $\phi$ to represent the fraction of high school graduates who go to college and $D$ to represent the upfront monetary costs of going to college.
+
+```{code-cell} ipython3
+D, ϕ = symbols('D phi', positive=True)
+```
+
+We can solve for by solving the equation below with regards to $\phi$
+
+$$
+PV_{highschool} = \phi PV_{college} - D
+$$
+
+so that 
+
+$$
+\phi =\frac{PV_{highschool}}{PV_{college} - D}
+$$
+
+- When $\phi$ is greater than 1, the person will choose to go to work.
+
+- When $\phi$ is less than 1, the person will choose to go to college.
+
+- When $\phi$ is equal to 1, the person is indifferent between going to college and going to work.
+
+Let's solve for $\phi$ using SymPy
+
+```{code-cell} ipython3
+indifference = Eq(PV_hs.args[1][0], 
+                  ϕ*PV_c.args[1][0] - D)
+indiff_sol = solve(indifference, ϕ)[0]
+indiff_sol
+```
+
+Using `Lambda` function, we can specify the parameters of the equation
+
+```{code-cell} ipython3
+lambda_indiff = Lambda((R, wh0, wc0, γ_c, γ_h, D, T), 
+                indiff_sol)
+lambda_indiff
+```
+
+We use the following parameters to compute the value for $\phi$
+
+```{code-cell} ipython3
+R_value = 1.05
+γ_c_value, γ_h_value = 1.01, 1.01
+wh0_value = 1
+wc0_value = 2
+D_value = 10
+T_value = 40
+
+lambda_indiff(R_value, wh0_value, wc0_value, 
+              γ_c_value, γ_h_value, D_value, 
+              T_value)
+```
+
+Under this setting, we find college is more attractive to the person as $\phi < 1$.
+
+Note that we can also use the `subs` method to substitute the values of the parameters
+
+```{code-cell} ipython3
+indiff_sol.subs({R: R_value, 
+                wh0: wh0_value, 
+                wc0: wc0_value, 
+                γ_c: γ_c_value, 
+                γ_h: γ_h_value, 
+                D: D_value, 
+                T: T_value})
+```
+
+It is also possible to use the [`evalf`](https://docs.sympy.org/latest/modules/evalf.html) method to get floating-point approximations
+
+```{code-cell} ipython3
+indiff_sol.evalf(subs={R: R_value, 
+                      wh0: wh0_value, 
+                      wc0: wc0_value, 
+                      γ_c: γ_c_value, 
+                      γ_h: γ_h_value, 
+                      D: D_value, 
+                      T: T_value})
+```
+
+Note how precision changes using different methods.
```

Comment:
Perhaps this could be:

It is useful to be aware that precision can be different depending on the method used to evaluate a function or solve an expression. 

We encourage readers to [read more here ]() 

With a link to the SymPy library. Is there a discussion on precision there we can point to?

Reply:
Many thanks @mmcky.

I think I will remove the second sentence as the documentation for `evalf` is pretty clear about the nature of floating point approximation.